File: app/src/main/java/com/lburne/bounded/data/model/Deck.kt
package com.lburne.bounded.data.model

import java.util.UUID

data class Deck(
    val deckId: String = UUID.randomUUID().toString(),
    val name: String = "",
    val format: DeckFormat = DeckFormat.CONSTRUCTED, // Updated default
    val legendCardId: String? = null,
    val championCardId: String? = null,
    val coverCardUrl: String? = null,
    val cardCount: Int = 0, // <--- NEW FIELD
    val lastModified: Long = System.currentTimeMillis()
)

File: Web App/src/models/deck.ts
export type DeckFormat = 'CONSTRUCTED';

export interface Deck {
    deckId: string;
    name: string;
    format: DeckFormat;
    legendCardId: string | null;
    championCardId: string | null;
    coverCardUrl: string | null;
    mainCount: number;
    sideCount: number;
    lastModified: number;
}

export interface DeckEntry {
    deckId: string;
    cardId: string;
    quantity: number;
    isSideboard: boolean;
}
